src/storage-client/src/controller.rs

```diff
@@ -1785,9 +1951,109 @@ where
         Ok(())
     }
 
-    /// Acquires persist handles for the given `shard`. This will `halt!` the
-    /// process if we cannot successfully acquire a critical handle with our
-    /// current epoch.
+    /// Returns IDs for all storage collections that the given
+    /// `collection_description` depends on.
+    ///
+    /// TODO: @sean: This is where the remap shard would slot it.
+    fn get_collection_dependencies(
+        &self,
+        collection_description: &CollectionDescription<T>,
+    ) -> Vec<GlobalId> {
+        let mut result = Vec::new();
+
+        if let DataSource::Ingestion(ingestion) = &collection_description.data_source {
+            match &ingestion.desc.envelope {
+                SourceEnvelope::Debezium(envelope_debezium) => {
+                    let tx_metadata_topic = envelope_debezium.dedup.tx_metadata.clone();
+                    if let Some(tx_input) = tx_metadata_topic {
+                        result.push(tx_input.tx_metadata_global_id);
+                    }
+                }
+                // NOTE: We explicitly list envelopes instead of using a catch all to
+                // make sure that we change this when adding/removing and envelope.
+                SourceEnvelope::None(_) | SourceEnvelope::Upsert(_) | SourceEnvelope::CdcV2 => {
+                    // No storage dependencies.
+                }
+            }
+        }
+
+        result
+    }
+
+    /// Install read capabilities on the given `storage_dependencies` and return
+    /// the frontier at which we are guaranteed to be able to read from all
+    /// dependencies.
+    fn install_read_capabilities(
```

Comment:
I notice this is called for both `create_collections` and `create_exports`, but its mate `remove_read_capabilities` is only called for sinks. Why's that?